### ISSUE TITLE
Hardly detected traffic signs

### ISSUE THREAD

**Issue by GITNE** [mapillary/mapillary_issues#2937]
In response to @tobias-o latest blog post and before you retrain the system and run any expensive update operation on the database, it might be helpful for you to identify certain caveat traffic signs which constantly slip through detection but are equivalent in semantics and belong to the same class. The following hints are just from my personal subjective observations, so I may be totally wrong.

The Swedish B1 and Polish A-7 yield signs are such candidates. They almost never get detected. I suppose this is because all Swedish and Polish triangular traffic signs have yellow backgrounds instead of white. 

Sign ID in OSM notation | NL:B6 | DE:205 | PL:A-7 | SE:B1
------------------------|-------|--------|--------|------
  | ![NL:B6](https://upload.wikimedia.org/wikipedia/commons/f/f1/Nederlands_verkeersbord_B6.svg "B6: Verleen voorrang aan bestuurders op de kruisende weg") | ![DE:205](https://upload.wikimedia.org/wikipedia/commons/6/6f/Zeichen_205_-_Vorfahrt_gew%C3%A4hren%21_StVO_1970.svg "Verkehrszeichen 205: „Vorfahrt gewähren“") | ![PL:A-7](https://upload.wikimedia.org/wikipedia/commons/c/c8/Znak_A-7.svg "Znak A-7: „ustąp pierwszeństwa”") | ![SE:B1](https://upload.wikimedia.org/wikipedia/commons/e/e0/Sweden_road_sign_B1.svg "B1: Väjningsplikt")

The problem I see here is that although there is a verification project running for the Dutch B6 yield sign—strangely enough with some German and UK yield signs sprinkled in here and there—the results will not improve detection of the Swedish B1 or Polish A-7 sign (and potentially others too) because by design this type of verification only improves true positive and lowers false positives rates. False negatives are completely out of scope. In other words, the resulting training material only improves on *more of the same*, not on *what should or should not be*. But, maybe I am missing here something.

---------------------------------------------------------------------------------------------------------------------------
This sign is almost never detected, although it is quite common. :disappointed: I do not know why. No verification project exists for this sign.

Sign ID in OSM notation | DE:250 | PL:B-1
------------------------|-------|--------
  | ![DE:250](https://upload.wikimedia.org/wikipedia/commons/4/46/Zeichen_250_-_Verbot_f%C3%BCr_Fahrzeuge_aller_Art%2C_StVO_1970.svg "Verkehrszeichen 250: „Verbot für Fahrzeuge aller Art“") | ![PL:B-1](https://upload.wikimedia.org/wikipedia/commons/e/ed/Znak_B-1.svg "Znak B-1: „zakaz ruchu w obu kierunkach”")

---------------------------------------------------------------------------------------------------------------------------
The German oneway sign 220-20 is almost never detected and should fall into the `regulatory--one-way-right`/`left--g1` class. Instead, white arrows on German parking signs which denote the beginning and end of parking zones, and are actually just a sub-element of the sign, are often falsely detected for `regulatory--one-way-right`/`left--g1` class signs (or Swedish E16-1 oneway sign).

Sign ID in OSM notation | DE:220-20 | SE:E16-1 | DE:314-20
------------------------|-----------|----------|----------
  | ![DE:220-20](https://upload.wikimedia.org/wikipedia/commons/9/94/Zeichen_220-20_-_Einbahnstra%C3%9Fe%2C_rechtsweisend%2C_StVO_2017.svg "Verkehrszeichen 220-20: „Einbahnstraße − rechtsweisend“") | ![SE:E16-1](https://upload.wikimedia.org/wikipedia/commons/8/85/Sweden_road_sign_E16-1-1.svg "E16-1: Enkelriktad trafik") | ![DE:314-20](https://upload.wikimedia.org/wikipedia/commons/3/38/Zeichen_314-20_-_Parkplatz_%28Ende%29%2C_StVO_1992.svg "314-20 „Parken, Ende (Aufstellung rechts) oder Anfang (Aufstellung links)“")

**Comment by tobias-o** [comment]
Hi @GITNE!

Thanks for your message. I'm happy to tell you that we address all of your mentioned issues in the current release.

ad yield signs:

Based on our validation data we are much better now in detecting the yellow yield signs. See [1] for an example. The yield sign was not detected with the old detector, now it works (including a lot of the small signs)

ad road closed sign:

This one is tricky to detect as a lot of other structures (e.g. the letter `o` in bill boards, car back lights at night) look similar.
With the high-level understanding of the image content we gain from our semantic segmentation we can remove most of those candidates.

ad one way:

We defined a traffic sign taxonomy to better represent "signs in signs". In the last step of our new pipeline we detect and handle this situation.

[1] https://www.mapillary.com/app/?lat=55.603245&lng=12.981916&z=17&pKey=pnPEXlmr_AOnAC_tJxm68Q&focus=photo&signs=true

**Comment by Etua** [comment]
@tobias-o As I can see not one A-7 sign present in my sequences is marked. Maybe it would be good to make a new verification challenge in order to teach detector a little bit more?

**Comment by tobias-o** [comment]
@Etua did you upload the images recently? Because all newly uploaded images get processed with the new method already.
We are reprocessing older uploads with full speed, but it might still take a while until all 300M+ images received the update

**Comment by Etua** [comment]
@tobias-o Yes, I uploaded them mostly yesterday and two days ago so I assume they have been processed with the new method and aside of few minor mismatches like recognizing building roof as traffic lights sign, the biggest issue is lack of A-7 supposedly may be useful in cases like navigation.

**Comment by tobias-o** [comment]
@Etua can you please send me the image key of an affected image?

**Comment by Etua** [comment]
@tobias-o For example in here the street sign should be pretty easy to recognize: https://www.mapillary.com/map/im/f4RxVqBMmNCkOPUl8tsKBw

**Comment by GITNE** [comment]
> I'm happy to tell you that we address all of your mentioned issues in the current release.

@tobias-o This is great! However, my impression is similar to @Etua’s. Can you provide the exact date when the latest detection mechanism went live, so that I can review my latest sequences? Additionally, I am going to capture a bunch of new sequences which should have some of the corner cases mentioned here.

**Comment by tobias-o** [comment]
@Etua thanks for the examples. For the first link, the sign gets detected, but its confidence score is below the threshold for actually putting it into the database. We have to set this threshold quite high because signs without border and content are sometimes hard to distinguish from random objects.

I'll have to do more checking on the second link. I issued a reprocessing then the sign appeared. Usually we detect missing processing and reprocess after 24h. This time I did the reprocessing manually before the time was up.
Also note that this release only affects traffic sign detection, not triangulation. We constantly try to improve this too though.

@GITNE we use the new pipeline for some time now, but the latest networks were put in place on 20th June, 9AM CEST
